### PR TITLE
Introduce Enum docs

### PR DESCRIPTION
**Pull request by vasekboch**

### CODE REVIEW

File: docs/enum.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+> In computer programming, an enumerated type (also called enumeration or enum)
+> is a data type consisting of a set of named values called elements, members
+> or enumerators of the type. The enumerator names are usually identifiers that
+> behave as constants in the language. A variable that has been declared as
+> having an enumerated type can be assigned any of the enumerators as a value.
+> In other words, an enumerated type has values that are different from each
+> other, and that can be compared and assigned, but which do not have any
+> particular concrete representation in the computer's memory; compilers and
+> interpreters can represent them arbitrarily.
```
Comment (jost125) [inline docs/enum.md]: `but which do not have any particular concrete representation in the computer's  memory; compilers and interpreters can represent them arbitrarily` 

This part is not true about our enums. Should we keep it?

File: docs/enum.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+[Wikipedia](http://wikipedia.org/wiki/Enumerated_type)
+
+## Usage
+
+We can define our Enum extending abstract Enum class. All possible values are defined via constants.
```
Comment (jost125) [inline docs/enum.md]: We should add a section mentioning, that enum values should be of type `string` or `int`.

File: docs/enum.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+	public static function GREEN(): Color {
+		return Color::create(self::GREEN);
+	}
+}
+
+$redColor = Color::RED();
+```
+
+All instances of same Enum value are eqal.
```
Comment (jost125) [inline docs/enum.md]: We should explicitly mention, that this returns same references.

File: docs/enum.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+	->mapKeys(fn(string $color): string => strtolower($color));
+
+// bool(true)
+var_dump($colorReverseMap->get('blue')->isDefined());
+
+// string(4) "EULB"
+var_dump($colorReverseMap->get('blue')->getUnsafe());
+```
+
+#### Complementary List
```
Comment (jost125) [inline docs/enum.md]: It is not complementary list (complementary means something different) it is `List Complement`. See https://en.wikipedia.org/wiki/Complement_(set_theory)

**Review by jost125 — CHANGES_REQUESTED** [review]

File: docs/enum.md (hunk 1 above)
Comment (honzatrtik) [reply docs/enum.md]: Maybe something like 
> Enum is type with values from finite closed set (think of boolean type, which has exactly two possible values - true & false)


Not sure if mentioning memory & compilers make it more clear ;-)

File: docs/enum.md (hunk 3 above)
Comment (honzatrtik) [reply docs/enum.md]: Maybe with code example demonstrating that `Color::RED() === Color::RED()`

Comment (honzatrtik) [reply docs/enum.md]: Ignore me, its already there ;-)

File: docs/enum.md (hunk 1 above)
Comment (jost125) [reply docs/enum.md]: :+1:

Comment (vasekboch) [reply docs/enum.md]: I was looking for some citation. The best I found was wikipedia.. I know its not perfect I will look for better source. But the Honza's definiton is good and simple :)

**Review by jost125 — APPROVED** [review]

**Pull request merged**